addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/handler/EvohomeHeatingZoneHandler.java

```diff
@@ -78,16 +90,118 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                 if (EvohomeBindingConstants.ZONE_SET_POINT_CHANNEL.equals(channelId)
                         && command instanceof DecimalType) {
                     double newTemp = ((DecimalType) command).doubleValue();
-                    if (newTemp == CANCEL_SET_POINT_OVERRIDE) {
+                    // Get a local copy of schedule when needed, saves on api calls
+                    DailySchedules schedule = getDailySchedules();
+                    double expectedTemp = getCurrentSetPoint(schedule);
+                    if (newTemp == CANCEL_SET_POINT_OVERRIDE || newTemp == expectedTemp) {
                         bridge.cancelSetPointOverride(getEvohomeThingConfig().id);
+                        updateState(EvohomeBindingConstants.ZONE_SET_POINT_CHANNEL, new DecimalType(expectedTemp));
+                        return;
                     } else if (newTemp < 5) {
                         newTemp = 5;
                     }
                     if (newTemp >= 5 && newTemp <= 35) {
-                        bridge.setPermanentSetPoint(getEvohomeThingConfig().id, newTemp);
+                        int overrideMode = this.getOverrideMode();
+                        switch (overrideMode) {
+                            case EvohomeOverrideModeConstants.TEMPORARY_OVERRIDE_SCHEDULE_NEXT:
+                                bridge.setTemporarySetPoint(getEvohomeThingConfig().id, newTemp,
+                                        getNextSetPointChange(schedule));
+                                break;
+                            case EvohomeOverrideModeConstants.TEMPORARY_OVERRIDE_TIME_ADD:
+                                bridge.setTemporarySetPoint(getEvohomeThingConfig().id, newTemp,
+                                        LocalDateTime.now().plusMinutes(this.getOverrideTime()));
+                                break;
+                            default:
+                                bridge.setPermanentSetPoint(getEvohomeThingConfig().id, newTemp);
+                                break;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private DailySchedules getDailySchedules() {
+        return this.getEvohomeBridge().getZoneSchedule("temperatureZone", this.getId());
+    }
+
+    private LocalDateTime getNextSetPointChange() {
+        return getNextSetPointChange(this.getDailySchedules());
+    }
+
+    private LocalDateTime getNextSetPointChange(DailySchedules schedule) {
+        if (schedule != null) {
+            int currentWeekDay = LocalDateTime.now().getDayOfWeek().getValue() - 1;
+            logger.debug("Current Weekday: " + currentWeekDay);
+            LocalDateTime nextSetPointChange = null;
+            for (DailySchedule daily : schedule.getSchedules()) {
+                // handle wrap around
+                int dayDelta = 0;
+                if (daily.getWeekday() < currentWeekDay) {
```

Comment:
This could do with some comment. I think this deals with the fact that if you get a weekday back that is before 'today' you make it so that is resembles next week?

Reply:
Clarified